irs/src/actions/actionCreators.js

```diff
@@ -2,31 +2,52 @@ import * as types from './actionTypes';
 import axios from '../axios/axios';
 
 export const login = (name, password) => dispatch => {
-  dispatch(loadingOn());
-  axios()({
-    method: 'post',
-    url: 'https://international-rural-school.herokuapp.com/api/auth/login',
-    headers: {},
-    data: {}
+  dispatch({
+    type: types.LOADING_ON
   })
-    .then(res => dispatch({ type: types.LOGIN, payload: res.data.payload }))
+  axios.post('https://international-rural-school.herokuapp.com/api/auth/login', {username: name, password: password})
     .then(res => {
+      dispatch({ type: types.LOGIN, payload: res.data.payload });
       dispatch({
         type: types.ADD_INFO,
         payload: {
           username: name,
           password: password
         }
       });
-      dispatch(loadingOff());
+      dispatch({
+        type: types.LOADING_OFF
+      })
     })
-
     .catch(err => {
-      dispatch(console.log(err));
-      dispatch(loadingOff());
+      console.log(err);
+      dispatch({
+        type: types.LOADING_OFF
+      });
     });
 };
 
+export const register = (name, password) => dispatch => {
+  console.log('registering', name, password)
+  dispatch({
+    type: types.REGISTERING
+  })
+  axios.post('https://international-rural-school.herokuapp.com/api/auth/register', {username: name, password: password})
```

Comment:
when Josh has the backend filled out, he'll let you know how to adjust this, just remember that you need to note the "permissions level" of the user as well to denote whether they're an admin or board member